docs/user-guide.md

```diff
@@ -85,6 +85,7 @@ Usage of kube-router:
       --run-firewall                                  Enables Network Policy -- sets up iptables to provide ingress firewall for pods. (default true)
       --run-router                                    Enables Pod Networking -- Advertises and learns the routes to Pods via iBGP. (default true)
       --run-service-proxy                             Enables Service Proxy -- sets up IPVS for Kubernetes Services. (default true)
+      --runtime-endpoint string                       Path to CRI compatible container runtime socket (used for DSR mode). Currently known working with containerd.
```

Comment:
@ep4eg sorry for the late review. I was wondering cant we just keep this generic like `cri-runtime`, and then we can infer runtime socket path etc internally? Or even avoid it totally if `podObj.Status.ContainerStatuses[0].ContainerID` can give us the details?

Reply:
Well, we can hardcode the default paths and select one based on what we get from `podObj.Status.ContainerStatuses[0].ContainerID`, however in this case we can't handle custom paths and runtimes other than containerd and cri-o. In the current design, we don't care what runtime is used, we just rely on CRI.

Reply:
So we want to avoid adding additional flags unless abolutley necessary, particualry something that can inferred or hardcoded.

>  In the current design, we don't care what runtime is used, we just rely on CRI.

My question is cant we infer? Also if we have hybrid CRI nodes across the clusters, then running kube-router as daemonset would be problem?

Reply:
Overall I would agree with having less flags if we can help it. I know one of the biggest things that drove us to use kube-router after evaluating it was how simple it made setup as compared to Calico and some of the other options out there. We should try to keep configuring it as simple as possible if we can.

Right now Kubernetes only officially supports Docker, CRI-O, and conainerd (https://kubernetes.io/docs/setup/production-environment/container-runtimes/). Docker support is already handled. If there are reliable defaults for the socket paths for CRI-O and containerd I would say we should just hard code them and remove the flag. If, in the future, a bunch of container runtimes come on to the scene and Kubernetes wants to support them, then we can add back the socket path at that point.

Reply:
I can't see any entropy in adding the flag but ok if you think it's better to hardcode, let's hardcode.

Reply:
I could see some logic in having hardcoded defaults, but not in removing the ability to configure the CRI runtime socket path here. The socket path is an implementation detail dependent on how the cluster was configured, and whilst there are common paths used for both containerd and CRI-O, these are not guaranteed.

Reply:
Ok, i see your point @axeal  I dont fully understand the context for `--container-runtime-endpoint` option of kubelet. Are you saying one can use containerd/CRI-O but still endpoint can be still different than what is hardcoded due to this option?

Also as I raised earlier if some one uses different run times in the clusters, specifiying it in daemonset does not make sense.

Reply:
@murali-reddy Yup, so `/run/containerd/containerd.sock` and `/var/run/crio/crio.sock` are common 'defaults' for the CRI runtime socket paths for containerd and cri-o respectively. However, these locations are configureable and can vary cluster to cluster, based on how the user/tooling set up the runtime. This PR currently sets the containerd default to `/var/run/containerd/containerd.sock`, which I assume was the path in @ep4eg's test environment and I subscribed to this PR as I'm interested in using DSR in a k3s cluster, for which the containerd socket path is `/run/k3s/containerd/containerd.sock`. Understand the point re. mixed runtimes, but that's an edge-case which I think is probably outside scope of the PR here. From what I know I don't think it's possible to infer that either, as the CRI socke path is a configuration detail of the kubelet and not exposed in a Kubernetes resource.

Reply:
@axeal thanks for the explanation. If there is a way we would have liked to avoid extra flag, but looks like not much choice. @ep4eg apologies for the back and forth requests, please revert back to your last commit to keep the flag

Reply:
done